### ISSUE TITLE
`no-node-access` false-positive warning on object property with `select` name

### ISSUE THREAD

**Issue by AlexKrupko** [testing-library/eslint-plugin-testing-library#1032]
### Have you read the Troubleshooting section?

Yes

### Plugin version

v7.5.3

### ESLint version

v9.29.0

### Node.js version

v23.7.0

### Bug description

After upgrading to version 7.5.3, the `no-node-access` warning appears on object property with `select` name

### Steps to reproduce

```ts
const ui = {
    select: selector.byRole('combobox', {name: 'Test label'}),
};

test('...', () => {
    // ......
    const select = ui.select.get();
    expect(select).toBeVisible();
    expect(select).toHaveClass(selectClasses.select);
});
```

Both `ui.select.get()` and `selectClasses.select` are warned. 

### ESLint configuration

--

### Rule(s) affected

no-node-access

### Do you want to submit a pull request to fix this bug?

No

**Comment by goodallt** [comment]
I am also experiencing this issue with a custom function we have implemented with the name `.select`

**Comment by ravicious** [comment]
In a similar fashion, it reports every callsite that uses [react-select-event](https://www.npmjs.com/package/react-select-event). Already reported in https://github.com/testing-library/eslint-plugin-testing-library/issues/1024#issuecomment-2965673059 but I wanted to highlight it here as well.

```tsx
await selectEvent.select(screen.getByLabelText('Version'), 'v6');
```

https://github.com/gravitational/teleport/blob/f3d51b360925c99f6332ff08415d44e058ebd881/web/packages/teleport/src/Roles/RoleEditor/StandardEditor/MetadataSection.test.tsx#L91

**Comment by Belco90** [comment]
This will be fixed by #1033

**Comment by anomiex** [comment]
While the specific example given above seems fixed, I'm still seeing a lot of false positives in my code. Trimmed down to a minimal-ish example, that code could look something like this:
```js
import { render, screen } from '@testing-library/react';
import { createRegistry } from '@wordpress/data';

test('...', () => {
    const registry = createRegistry();
    expect( registry.select( 'store' ) ).not.toBeUndefined();
});
```

Or even
```js
import { render, screen } from '@testing-library/react';

function createRegistry() {
    return {
        select: store => "ok",
    };
}

test('...', () => {
    const registry = createRegistry();
    expect( registry.select( 'store' ) ).not.toBeUndefined();
});
```

**Comment by joshkel** [comment]
We're also seeing false positives with 7.5.4. In our case, it's from [Knex](https://knexjs.org/) database calls in our tests:

```ts
    const result = await db.select().table('DataRecord');
```

(We could probably try and change our setup to strictly separate tests that reference the DOM from tests that reference the database then change our ESLint rules to only run eslint-plugin-testing-library on DOM tests, but that's an invasive change.)
